gammapy/region/core.py

```diff
@@ -0,0 +1,231 @@
+                * ``'exact'``: returns a value between 0 and 1 giving the
+                  fractional level of overlap of the pixel with the region.
+
+        Returns
+        -------
+        mask : `~numpy.ndarray`
+            A mask indicating whether each pixel is contained in the region.
+        slice_x, slice_y : `slice`
+            Slices for x and y which can be used on an array to extract the
+            same region as the mask.
+        """
+        raise NotImplementedError("")
+
+    def to_shapely(self):
+        """
+        Convert this region to a Shapely object.
+        """
+        raise NotImplementedError("")
+
+
+@six.add_metaclass(abc.ABCMeta)
+class SkyRegion(Region):
+    """
+    Base class for all regions defined in celestial coordinates
+    """
+
+    def __contains__(self, skycoord):
+        """
+        Checks whether a position or positions fall inside the region.
+
+        Parameters
+        ----------
+        skycoord : `~astropy.coordinates.SkyCoord`
+            The position or positions to check
+        """
+        raise NotImplementedError("")
+
+    def area(self):
+        """
+        Returns the area of the region as a `~astropy.units.Quantity`.
+        """
+        raise NotImplementedError("")
+
+    def to_pixel(self, wcs, mode='local', tolerance=None):
+        """
+        Returns a region defined in pixel coordinates.
+
+        Parameters
+        ----------
+
+        wcs : `~astropy.wcs.WCS` instance
+            The world coordinate system transformation to assume
+
+        mode : str
+            Convering to pixel coordinates can be done with various degrees
+            of approximation, which can be set with this option. Possible
+            values are:
+
+            * `'local'`: assume that the field of view is small and that
+              pixels are square, so that e.g. a circle in sky coordinates
+              would be a circle in pixel coordinates. This is the fastest and
+              most commonly used for e.g. photometry.
+
+            * `'affine'`: approximate any deviations from the 'local'
+              assumption by an affine transformation, e.g. a circle would
+              become a rotated ellipse.
+
+            * `'full'`: return an arbitrarily complex polygon in pixel
+              coordinates that represents the full level of distortion due to
+              the conversion from world to pixel coordinates. The degree of
+              exactness can be controlled by the ``tolerance`` argument.
+
+        tolerance : `~astropy.units.Quantity`
+            The tolerance for the ``'full'`` mode described above.
+        """
+        raise NotImplementedError("")
+
+
+class RegionList(list):
+    """List of regions"""
+
+    def write(self, filename, format='ds9'):
+        """Write list of regions to file
+
+        Parameters
+        ----------
+        filename : str
+            Name of file to write
+        format : str
```

Comment:
Multiple-choice options are documented in docstrings like this:
```
parameter : {"choice1", "choice2"}
    A parameter
```

Also, it's better to have a method that generates string and then the method that writes to file just calls that one.
(I hate APIs where you have to write to temp files or in-memory file-like buffers even if you just want a string)